tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -755,7 +755,7 @@ static bool passCursorInfoForDecl(const ValueDecl *VD,
         PO.SkipIntroducerKeywords = true;
         PO.ArgAndParamPrinting = PrintOptions::ArgAndParamPrintingMode::ArgumentOnly;
         XMLEscapingPrinter Printer(RelDeclsStream);
-        if (BaseType) {
+        if (BaseType && RelatedDecl->getDeclContext() == BaseType->getAnyNominal()) {
```

Comment:
Imagine you have this:

    class Outer<T> { class Inner<U> { ... } ... }

You might have BaseType of Outer.Inner, and unqualified lookup might find a member of Outer. The member's DeclContext is a type context but it's not the right one for BaseType.

If you use all the information that unqualified lookup gives you, you will be able to find the right base type.